app/src/utils/helper.js

```diff
@@ -1,40 +1,59 @@
 import { NavigationActions, StackActions } from 'react-navigation';
 
-export function navigateToUri(navigation, uri, additionalParams) {
-    if (!navigation) {
+export function dispatchNavigateToUri(dispatch, nav, uri) {
+  const params = { uri };
+  if (nav && nav.routes && nav.routes.length > 0 && 'Main' === nav.routes[0].routeName) {
```

Comment:
Can you add more helper methods for detecting the state of routes?
This is a lot of fragile code that can quickly become unmanageable.